src/components/dom.ts

```diff
@@ -634,4 +634,28 @@ export default class Dom {
       right: left + rect.width,
     };
   }
+
+  /**
+   * Returns true if element supports selection
+   *
+   * @param {*} target - HTML element or string
+   *
+   * @returns {boolean}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public static supportsSelection(target: any): boolean {
```

Comment:
we already have the `canSetCaret` method. What is the difference between them?

Reply:
the logic looks similar except the case with contenteditable

Reply:
@neSpecc So what should we do?

Reply:
HTML inputs have 22 different types, according to [mdn docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types). `canSetCaret` checks for 8 of them which leaves 14. Out of those 14 only 5 support selection so the methods check for quite a different set of input types.

I totally get the idea of reusing code but I'm not quite sure if that's feasible here. Also separating them creates a bit more easily understandable code

Reply:
I think the reason of the `canSetCaret` existance is the same as that method. So maybe we could improve `canSetCart` and update its logic to cover more cases?

Reply:
Sure that might work, issue is that with current implementation `canSetCaret` return true for `contenteditable` but `contenteditable` doesn't support selection. I now basically changed `canSetCaret` implementation to the one I made for `supportsSelection` and thus dropped support for `contenteditable`.

It seems to work quite well as far as I checked since previously the `isNativeInput` was used for checking selection stuff and it doesn't support `contenteditable`. Would be glad to hear thoughts on this!